src/Controller/ControllerAbstract.php

```diff
@@ -36,162 +22,127 @@ abstract class ControllerAbstract implements IController
     /** @var IResponse */
     protected $response;
 
-    public function __construct()
-    {
-        Application::getInstance()->getDiContainer()->getViewDo()->clear();
-    }
-
-    public function setRequest(IRequest $request): void
+    public function setRequest(IRequest $request)
```

Comment:
A self typehint (with a $this docblock) would be great for these setters

Reply:
See interface